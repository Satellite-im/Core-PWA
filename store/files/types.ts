import { FileType, Folder } from '~/types/files/file'

export interface FilesState {
  tree: Array<FileType | Folder>
}
