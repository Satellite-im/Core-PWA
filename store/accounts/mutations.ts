import { AccountsState, RegistrationStatus } from './types'
import { User } from '~/libraries/Iridium/users/types'

const mutations = {
  setPin(state: AccountsState, pin: string) {
    state.pin = pin
  },
  setStorePin(state: AccountsState, storePin: boolean) {
    state.storePin = storePin
  },
  setRegistry(state: AccountsState, registry: boolean) {
    state.registry = registry
  },
  setPinHash(state: AccountsState, pinHash: string) {
    state.pinHash = pinHash
  },
  unlock(state: AccountsState, pin: string) {
    state.locked = false
    state.pin = pin
  },
  lock(state: AccountsState) {
    state.locked = true
  },
  setEncryptedPhrase(state: AccountsState, encryptedPhrase: string) {
    state.encryptedPhrase = encryptedPhrase
  },
  setPhrase(state: AccountsState, decryptedPhrase: string) {
    state.phrase = decryptedPhrase
  },
  setActiveAccount(state: AccountsState, activeAccountPubkey: string) {
    state.active = activeAccountPubkey
    state.initialized = true
  },
  setProfilePicture(state: AccountsState, image: string) {
    if (state.details) {
      state.details.photoHash = image
    }
  },
  setUserDetails(state: AccountsState, details: User) {
    state.details = {
      did: details.did,
      name: details.name,
      status: details.status,
      photoHash: details.photoHash,
    }
    state.welcomePopup = true
  },
  dismissWelcome(state: AccountsState) {
    state.welcomePopup = false
  },
  setUserPeerId(state: AccountsState, peerId: string) {
    if (state.details) {
      state.details = {
        ...state.details,
        peerId,
      }
    }
  },
  setRegistrationStatus(
    state: AccountsState,
    registrationStates: RegistrationStatus,
  ) {
    state.registrationStatus = registrationStates
  },
  setLastVisited(state: AccountsState, lastVisited: string) {
    state.lastVisited = lastVisited
  },
  setAdapter(state: AccountsState, adapter: string) {
    state.adapter = adapter
  },
  setEntropy(state: AccountsState, entropyMessage: string) {
    state.entropyMessage = entropyMessage
  },
}

export default mutations
