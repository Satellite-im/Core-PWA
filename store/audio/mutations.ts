import { AudioState } from './types'

const mutations = {
  mute(state: AudioState) {
    state.muted = !state.muted
  },
  toggleMute(state: AudioState) {
    state.muted = !state.muted
  },
  deafen(state: AudioState) {
    const isDeafened = state.deafened
    state.previousVolume = isDeafened ? state.previousVolume : state.volume
    state.volume = isDeafened ? state.previousVolume : 0
    state.deafened = !isDeafened
  },
  setVolume(state: AudioState, volume: Number) {
    state.previousVolume = volume
    state.volume = volume
  },
  setSystemVolume(state: AudioState, systemVolume: Number) {
    state.sounds.system = systemVolume
  },
  setInputVolume(state: AudioState, inputVolume: Number) {
    state.inputVolume = inputVolume
  },
  setSoundLevel(
    state: AudioState,
    sound: {
      inboundMedia: Number
      outboundMedia: Number
      system: Number
    },
  ) {
    state.sounds = sound
  },
}

export default mutations
