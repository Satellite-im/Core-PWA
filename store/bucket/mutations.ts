import { BucketState } from './types'
import { FilSystem } from '~/libraries/Files/FilSystem'

const mutations = {
  bucketInitialized(state: BucketState, status: boolean) {
    state.initialized = status
  },
  updateFileSystem(state: BucketState, fs: FilSystem) {
    state.fileSystem = fs
  },
  openDirectory(state: BucketState, name: string) {
    state.fileSystem.openDirectory(name)
  },
  createDirectory(state: BucketState, name: string) {
    state.fileSystem.createDirectory(name)
  },
  goBackToDirectory(state: BucketState, name: string) {
    state.fileSystem.goBackToDirectory(name)
  },
  createFile(state: BucketState, name: string) {
    state.fileSystem.createFile(name)
  },
}

export default mutations
