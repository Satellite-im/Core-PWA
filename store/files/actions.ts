import { Files } from '~/mock/files'

export default {
  handler: () => {},
  // @ts-ignore
  async fetchFiles({ commit }) {
    await new Promise((resolve) => setTimeout(resolve, 3000))
    commit('fetchFiles', Files)
    commit('load', 'files')
  },
}
