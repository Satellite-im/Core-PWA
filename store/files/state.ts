import { FilesState } from './types'

const InitalFilesState = (): FilesState => ({
  tree: [],
})

export default InitalFilesState
