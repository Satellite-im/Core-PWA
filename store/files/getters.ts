import Vue from 'vue'
import { GetterTree } from 'vuex'
import fuzzysort from 'fuzzysort'
import { FileRouteEnum, FileSortEnum } from '~/libraries/Enums/enums'
import { IridiumDirectory, IridiumItem } from '~/libraries/Iridium/files/types'
import { FilesState } from '~/store/files/types'
import { RootState } from '~/types/store/store'
import iridium from '~/libraries/Iridium/IridiumManager'

export interface FilesGetters {
  sortedItems(
    state: FilesState,
  ): (filteredFiles: IridiumItem[], route: FileRouteEnum) => IridiumItem[]
}

const getters: GetterTree<FilesState, RootState> & FilesGetters = {
  searchedItems:
    (state: FilesState) =>
    (items: IridiumItem[], searchFilter: string): IridiumItem[] => {
      if (!searchFilter) {
        return items
      }

      if (state.search.searchAll) {
        items = iridium.files.flat
      } else {
        const parentId = state.path[state.path.length - 1]?.id
        const parent = iridium.files.flat.find(
          (e) => e.id === parentId,
        ) as IridiumDirectory

        if (parent?.children) {
          items = parent.children
        }
      }

      const results = fuzzysort.go(searchFilter, items, {
        keys: ['name'],
      })
      Vue.delete(results, 'total')

      return results.map((e: { obj: {} }) => e.obj as IridiumItem)
    },

  sortedItems:
    (state: FilesState) =>
    (items: IridiumItem[], route: FileRouteEnum): IridiumItem[] => {
      const key = state.sort.category
      const searchAllActive = state.search.searchAll && state.search.value

      // If "SearchAll" is inactive, we only want to sort the items that are in the current directory
      if (!searchAllActive && state.path.length) {
        const parentId = state.path[state.path.length - 1]?.id
        const parent = iridium.files.flat.find(
          (e) => e.id === parentId,
        ) as IridiumDirectory

        if (parent?.children) {
          items = parent.children
        }
        // If set to "Recent", get 15 most recently edited files (no directories)
      } else if (route === FileRouteEnum.RECENT) {
        const recentItems = iridium.files.flat
          .filter((e) => !('children' in e))
          .sort((a, b) => b.modified - a.modified)
          .slice(0, 14)

        if (recentItems) {
          items = recentItems
        }
      }

      const collator = new Intl.Collator()
      // Sort the items by the selected key
      switch (key) {
        case FileSortEnum.SIZE:
          items.sort((a, b) => a.size - b.size)
          break
        case FileSortEnum.MODIFIED:
          items.sort((a, b) => b.modified - a.modified)
          break
        default:
          // Sort alphabetically
          items.sort((a, b) => collator.compare(a[key], b[key]))
      }

      // Reverse the sort if the selected key is descending
      if (!state.sort.asc) {
        items = items.reverse()
      }

      return items
    },
}

export default getters
