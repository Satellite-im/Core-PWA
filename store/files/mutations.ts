import { FileRouteEnum, FileSortEnum } from '~/libraries/Enums/enums'
import { IridiumFile, IridiumItem } from '~/libraries/Iridium/files/types'
import { FilesState } from '~/store/files/types'

const mutations = {
  addDownload(state: FilesState, name: string) {
    state.downloadList.push(name)
  },
  removeDownload(state: FilesState, name: string) {
    const index = state.downloadList.indexOf(name)
    if (index > -1) {
      state.downloadList.splice(index, 1)
    }
  },
  setCurrentUpload(state: FilesState, item?: FilesState['currentUpload']) {
    state.currentUpload = item
  },
  setRename(state: FilesState, item?: IridiumItem) {
    state.rename = item
  },
  setPath(state: FilesState, path: { id: string; name: string }[]) {
    state.path = path
  },
  setPreview(state: FilesState, file?: IridiumFile) {
    state.preview = file
  },
  setRoute(state: FilesState, route: FileRouteEnum) {
    state.route = route
  },
  // if current category, swap asc/desc. if different, change category
  setSort(state: FilesState, category: FileSortEnum) {
    state.sort =
      state.sort.category === category
        ? { category: state.sort.category, asc: !state.sort.asc }
        : { category, asc: true }
  },
  setStatus(state: FilesState, status: string) {
    state.status = status
  },
  toggleLayout(state: FilesState) {
    state.gridLayout = !state.gridLayout
  },
}

export default mutations
