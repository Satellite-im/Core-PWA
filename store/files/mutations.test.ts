import * as mutations from '~/store/files/mutations'

describe('mutations.default.addDownload', () => {
  test('0', () => {
    let result: any = mutations.default.addDownload(
      { downloadList: { push: () => 10 } },
      'Jean-Philippe',
    )
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    let result: any = mutations.default.addDownload(
      { downloadList: { push: () => 32 } },
      'Anas',
    )
    expect(result).toMatchSnapshot()
  })

  test('2', () => {
    let result: any = mutations.default.addDownload(
      { downloadList: { push: () => 256 } },
      'Edmond',
    )
    expect(result).toMatchSnapshot()
  })

  test('3', () => {
    let result: any = mutations.default.addDownload(
      { downloadList: { push: () => 16 } },
      'Pierre Edouard',
    )
    expect(result).toMatchSnapshot()
  })

  test('4', () => {
    let result: any = mutations.default.addDownload(
      { downloadList: { push: () => 32 } },
      'Edmond',
    )
    expect(result).toMatchSnapshot()
  })

  test('5', () => {
    let result: any = mutations.default.addDownload(
      { downloadList: { push: () => -Infinity } },
      '',
    )
    expect(result).toMatchSnapshot()
  })
})

describe('mutations.default.removeDownload', () => {
  test('0', () => {
    let result: any = mutations.default.removeDownload(
      { downloadList: { indexOf: () => 0, splice: () => 16 } },
      'Edmond',
    )
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    let result: any = mutations.default.removeDownload(
      { downloadList: { indexOf: () => 1, splice: () => 10 } },
      'Jean-Philippe',
    )
    expect(result).toMatchSnapshot()
  })

  test('2', () => {
    let result: any = mutations.default.removeDownload(
      { downloadList: { indexOf: () => 100, splice: () => 16 } },
      'Pierre Edouard',
    )
    expect(result).toMatchSnapshot()
  })

  test('3', () => {
    let result: any = mutations.default.removeDownload(
      { downloadList: { indexOf: () => -100, splice: () => 10 } },
      'Jean-Philippe',
    )
    expect(result).toMatchSnapshot()
  })

  test('4', () => {
    let result: any = mutations.default.removeDownload(
      { downloadList: { indexOf: () => 100, splice: () => 0 } },
      'Pierre Edouard',
    )
    expect(result).toMatchSnapshot()
  })

  test('5', () => {
    let result: any = mutations.default.removeDownload(
      { downloadList: { indexOf: () => NaN, splice: () => NaN } },
      '',
    )
    expect(result).toMatchSnapshot()
  })
})

describe('mutations.default.setSort', () => {
  test('0', () => {
    let result: any = mutations.default.setSort(
      {
        sort: {
          category:
            'The slim & simple Maple Gaming Keyboard from Dev Byte comes with a sleek body and 7- Color RGB LED Back-lighting for smart functionality',
          asc: 56784,
        },
      },
      'New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart',
    )
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    let result: any = mutations.default.setSort(
      {
        sort: {
          category:
            'The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design',
          asc: 'a1969970175',
        },
      },
      'New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart',
    )
    expect(result).toMatchSnapshot()
  })

  test('2', () => {
    let result: any = mutations.default.setSort(
      {
        sort: {
          category:
            'The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design',
          asc: 12345,
        },
      },
      'The Nagasaki Lander is the trademarked name of several series of Nagasaki sport bikes, that started with the 1984 ABC800J',
    )
    expect(result).toMatchSnapshot()
  })

  test('3', () => {
    let result: any = mutations.default.setSort(
      {
        sort: {
          category:
            "Boston's most advanced compression wear technology increases muscle oxygenation, stabilizes active muscles",
          asc: 'a1969970175',
        },
      },
      'The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design',
    )
    expect(result).toMatchSnapshot()
  })

  test('4', () => {
    let result: any = mutations.default.setSort(
      {
        sort: {
          category:
            'New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart',
          asc: 56784,
        },
      },
      'New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart',
    )
    expect(result).toMatchSnapshot()
  })

  test('5', () => {
    let result: any = mutations.default.setSort(
      { sort: { category: '', asc: Infinity } },
      '',
    )
    expect(result).toMatchSnapshot()
  })
})

describe('mutations.default.toggleLayout', () => {
  test('0', () => {
    let result: any = mutations.default.toggleLayout({
      gridLayout:
        'data:image/svg+xml;charset=UTF-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20version%3D%221.1%22%20baseProfile%3D%22full%22%20width%3D%22undefined%22%20height%3D%22undefined%22%3E%3Crect%20width%3D%22100%25%22%20height%3D%22100%25%22%20fill%3D%22grey%22%2F%3E%3Ctext%20x%3D%22NaN%22%20y%3D%22NaN%22%20font-size%3D%2220%22%20alignment-baseline%3D%22middle%22%20text-anchor%3D%22middle%22%20fill%3D%22white%22%3Eundefinedxundefined%3C%2Ftext%3E%3C%2Fsvg%3E',
    })
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    let result: any = mutations.default.toggleLayout({ gridLayout: '' })
    expect(result).toMatchSnapshot()
  })
})
