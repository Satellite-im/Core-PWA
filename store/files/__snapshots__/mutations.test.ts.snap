// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations.default.addDownload 0 1`] = `undefined`;

exports[`mutations.default.addDownload 1 1`] = `undefined`;

exports[`mutations.default.addDownload 2 1`] = `undefined`;

exports[`mutations.default.addDownload 3 1`] = `undefined`;

exports[`mutations.default.addDownload 4 1`] = `undefined`;

exports[`mutations.default.addDownload 5 1`] = `undefined`;

exports[`mutations.default.removeDownload 0 1`] = `undefined`;

exports[`mutations.default.removeDownload 1 1`] = `undefined`;

exports[`mutations.default.removeDownload 2 1`] = `undefined`;

exports[`mutations.default.removeDownload 3 1`] = `undefined`;

exports[`mutations.default.removeDownload 4 1`] = `undefined`;

exports[`mutations.default.removeDownload 5 1`] = `undefined`;

exports[`mutations.default.setSort 0 1`] = `undefined`;

exports[`mutations.default.setSort 1 1`] = `undefined`;

exports[`mutations.default.setSort 2 1`] = `undefined`;

exports[`mutations.default.setSort 3 1`] = `undefined`;

exports[`mutations.default.setSort 4 1`] = `undefined`;

exports[`mutations.default.setSort 5 1`] = `undefined`;

exports[`mutations.default.toggleLayout 0 1`] = `undefined`;

exports[`mutations.default.toggleLayout 1 1`] = `undefined`;
